server/stream.go

```diff
@@ -839,6 +839,28 @@ func (mset *stream) isLeader() bool {
 	return true
 }
 
+// Lock should be held
+func (mset *stream) stepDown() {
+	// cancel timer to create the source consumers if not fired yet
+	if mset.sourcesConsumerSetup != nil {
+		mset.sourcesConsumerSetup.Stop()
+		mset.sourcesConsumerSetup = nil
+	} else { // otherwise cancel the source consumers
```

Comment:
Should we qualify if needed here?

Reply:
Technically the stopSourceConsumers() could be always executed rather than only if mset.sourcesConsumerSetpup != nil.

But it is now removed along with all the other cleanup I had added initially.

Reply:
Added back